src/Adapter/Product/Pack/Repository/ProductPackRepository.php

```diff
@@ -28,16 +28,80 @@
 use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Product\Pack\Exception\ProductPackException;
 use PrestaShop\PrestaShop\Core\Domain\Product\Pack\ValueObject\PackId;
 use PrestaShop\PrestaShop\Core\Domain\Product\QuantifiedProduct;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
 use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
 use PrestaShopException;
+use Throwable;
 
-class ProductPackRepository
+class ProductPackRepository extends AbstractObjectModelRepository
 {
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     *
+     * @return array<array<string, string>>
+     *                             e.g [
+     *                             ['id_product' => '1', 'name' => 'Product name', 'reference' => 'demo15'],
+     *                             ['id_product' => '2', 'name' => 'Product name2', 'reference' => 'demo16'],
+     *                             ]
+     *
+     * @throws CoreException
+     */
+    public function getPackedProducts(ProductId $productId, LanguageId $languageId): array
+    {
+        $this->assertProductExists($productId);
+        $productIdValue = $productId->getValue();
+
+        try {
+            $qb = $this->connection->createQueryBuilder();
+            $qb->select('pack.id_product_item, pack.id_product_attribute_item, pack.quantity, packed.reference, language.name')
+                ->from($this->dbPrefix . 'pack', 'pack')
+                ->leftJoin('pack', $this->dbPrefix . 'product', 'product', 'product.id_product = pack.id_product_pack')
+                ->leftJoin('pack', $this->dbPrefix . 'product', 'packed', 'packed.id_product = pack.id_product_item')
+                ->leftJoin('pack', $this->dbPrefix . 'product_lang', 'language', 'pack.id_product_item = language.id_product')
+                ->where('pack.id_product_pack = :idProduct')
+                ->andWhere('language.id_lang = :idLanguage')
+                ->orderBy('pack.id_product_item', 'ASC')
+                ->setParameter('idProduct', $productId->getValue())
+                ->setParameter('idLanguage', $languageId->getValue());
+            $packedProducts = $qb->execute()->fetchAll();
+        } catch (Throwable $e) {
+            throw new CoreException(sprintf(
+                'Error occurred when fetching related products for product #%d',
+                $productIdValue
+            ));
```

Comment:
If the DB error is not logged here, I suggest to provide it as the previous exception to help maintenance:
```suggestion
            throw new CoreException(sprintf(
                'Error occurred when fetching related products for product #%d',
                $productIdValue,
                $e
            ));
```

Reply:
@saulaski  Are you sure that you want add it into the sprintf ?

Reply:
Absolutely not ! my mistake... GitHub comments are not the best IDE :)
I meant this instead:
```suggestion
            throw new CoreException(
                sprintf(
                    'Error occurred when fetching related products for product #%d',
                    $productIdValue
                ),
                0,
                $e
            );
```

Reply:
I agree, returning the previous exception seems like a good idea to help debugging